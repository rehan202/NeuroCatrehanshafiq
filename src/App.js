import React, { useState, useEffect } from 'react';
import logo from './logo.svg';
import './App.css';
import Q2 from "./Q2/q2.js";
import Q3 from "./Q3/q3.js";

function callBackFunctionButton() {
  alert("I am button and I am clicked.");
  console.log('I am button and I am clicked.');
}

function callBackFunctionTextBox(p) {
  alert("Hello! I am an parent component. Child Component changed and value passed to me and value is = " + p);
  console.log("Hello! I am an parent component. Child Component changed and value passed to me and value is = " + p);
}

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h1>React</h1>
        <img src={logo} className="App-logo" alt="logo" />
        <Q2
          isBlock="false"
          isDisabled="false"
          isDanger="false"
          name="My Button"
          onClick={callBackFunctionButton}
        />

        <br />

        <Q3
          value="Please enter your name."
          onClick={callBackFunctionTextBox}
        />
      </header>
    </div>
  );
}

export default App;
