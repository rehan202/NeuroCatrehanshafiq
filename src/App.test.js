import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { render, cleanup } from '@testing-library/react';

afterEach(cleanup);

it("renders App snapshot test", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
