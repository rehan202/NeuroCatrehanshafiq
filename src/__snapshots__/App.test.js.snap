// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App snapshot test 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h1>
        React
      </h1>
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <div>
        <button
          data-testid="myButton"
          style="display: inline; background-color: Green;"
        >
          My Button
        </button>
      </div>
      <br />
      <div>
        <input
          data-testid="myTextbox"
          type="text"
          value="Please enter your name."
        />
      </div>
    </header>
  </div>
</DocumentFragment>
`;
